src/Pagerfanta/Adapter/MappingAdapter.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Pagerfanta package.
+ *
+ * (c) Pablo Díez <pablodip@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Pagerfanta\Adapter;
+
+class MappingAdapter implements AdapterInterface
+{
+    protected $innerAdapter;
+    protected $callback;
```

Comment:
I suggest making these private. It makes maintenance much easier to deal with semver (protected stuff are subject to backward compatibility too)